Build.hs

```diff
@@ -0,0 +1,967 @@
+        , "  " ++ src </> "utils/unlit"
+        , "  " ++ src </> "libraries/array"
+        , "  " ++ src </> "libraries/binary"
+        , "  " ++ src </> "libraries/bytestring"
+        , "  " ++ src </> "libraries/containers/containers"
+        , "  " ++ src </> "libraries/deepseq"
+        , "  " ++ src </> "libraries/directory/"
+        , "  " ++ src </> "libraries/exceptions"
+        , "  " ++ src </> "libraries/file-io/"
+        , "  " ++ src </> "libraries/filepath/"
+        , "  " ++ src </> "libraries/mtl"
+        , "  " ++ src </> "libraries/os-string/"
+        , "  " ++ src </> "libraries/parsec"
+        , "  " ++ src </> "libraries/pretty/"
+        , "  " ++ src </> "libraries/process/"
+        , "  " ++ src </> "libraries/semaphore-compat"
+        , "  " ++ src </> "libraries/text"
+        , "  " ++ src </> "libraries/time"
+        , "  " ++ src </> "libraries/transformers"
+        , "  " ++ src </> "libraries/unix/"
+        , "  " ++ src </> "libraries/Win32/"
+        , "  " ++ src </> "libraries/Cabal/Cabal-syntax"
+        , "  " ++ src </> "libraries/Cabal/Cabal"
+        , "  https://github.com/haskell/alex/archive/refs/tags/v3.5.2.0.tar.gz"
+        , ""
+        , "benchmarks: False"
+        , "tests: False"
+        , "allow-boot-library-installs: True"
+        , "active-repositories: :none"
+        , ""
+        , "package *"
+        , "  library-vanilla: True"
+        , "  shared: False"
+        , "  executable-profiling: False"
+        , "  executable-dynamic: False"
+        , "  executable-static: False"
+        , ""
+        , "package ghc-internal"
+             -- FIXME: make our life easier for now by using the native bignum backend
+        , "  flags: +bignum-native"
+        , ""
+        , "package text"
+             -- FIXME: avoid having to deal with system-cxx-std-lib fake package for now
+        , "  flags: -simdutf"
+        , ""
+        ] ++ rts_options
+
+
+  let boot_libs_env = dst </> "boot-libs.env"
+  let build_boot_cmd = runCabal cabal
+        [ "install"
+        , "--lib"
+        , "--package-env=" ++ boot_libs_env
+        , "--force-reinstalls"
+        , "-v3"
+        , "--project-file=" ++ cabal_project_bootlibs_path
+        , "--with-compiler=" ++ ghcPath ghc
+        , "--with-hc-pkg=" ++ ghcPkgPath ghcpkg
+        , "--ghc-options=\"-ghcversion-file=" ++ ghcversionh ++ "\""
+        , "--builddir=" ++ build_dir
+        , "-j"
+
+          -- targets
+        , "rts"
+        , "ghc-internal"
+        , "base"
+          -- shallow compat packages over ghc-internal
+        , "ghc-prim"
+        , "ghc-bignum"
+        , "integer-gmp"
+        -- , "template-haskell" -- FIXME: adding this fails because cabal runs
+        -- ghc for some reason, without having the rts built (obviously, we're
+        -- building it!), hence it crashes because ghc can't find
+        -- `ghcversion.h`... which is the in the rts package it hasn't built yet
+        -- ffs!
```

Comment:
@andreabedini This is quite annoying: cabal-install shouldn't run GHC (apparently it is to compile some .c file to test something, I guess). It shouldn't do this if there is no rts in the package db and no `-ghcversion-file` passed on the command line. Any hint of where I should look at tomorrow to disable this?

Reply:
Let's pair up on this

Reply:
The failure is unrelated though, am I right?
```
No template specified, and template-hsc.h not located.
```

Reply:
Oh I didn't get that failure locally with a non-clean build. I'm trying to reproduce it.

Reply:
Ok so it doesn't fail locally with a clean build.

Reply:
![screenshot-20250204@095058](https://github.com/user-attachments/assets/14bcb645-a441-49bb-a35f-fd0009a3b899)

Reply:
I've made the script copy `template-hsc.h` into GHC's libdir and enabled `IN_GHC_TREE`. This is so ugly, the file is 138 line long, we could just embed it and use the embedded one instead of failing like this.

Reply:
It didn't work. I'll look into what is returned by `getExecutablePath`

Reply:
It works now

Reply:
Back to the original issue with cabal running ghc. I'll look into it

Reply:
So the bogus GHC call seems to be happening from `Distribution.Simple.GHC.Internal.configureLd'`: it uses GHC to build a dumb `.o` that can be used to test if `ld` supports `-x`

Reply:
So I've tried to replace the call the `ghcProg` command with a call to `runGHC` with the "configured" GHC program... But now it fails earlier because `ghcversion.h` is missing.

I'll just disable the `ld -x` check, add a FIXME and move on.

Reply:
I've allowed the use of GHC without an rts and it works now! We can even enable template-haskell apparently according to CI (locally it failed for other reasons :'()